tests/test_fakes.py

```diff
@@ -1542,6 +1543,54 @@ def test_use_invalid_schema(_fakesnow: None):
         )
 
 
+# Snowflake SQL variables: https://docs.snowflake.com/en/sql-reference/session-variables#using-variables-in-sql
+#
+# Variables are scoped to the session (Eg. The connection, not the cursor)
+# [x] Simple scalar variables: SET var1 = 1;
+# [x] Unset variables: UNSET var1;
+# [x] Simple SQL expression variables: SET INCREMENTAL_DATE = DATEADD( 'DAY', -7, CURRENT_DATE());
+# [x] Basic use of variables in SQL using $ syntax: SELECT $var1;
+# [ ] Multiple variables: SET (var1, var2) = (1, 'hello');
+# [ ] Variables set via 'properties' on the connection https://docs.snowflake.com/en/sql-reference/session-variables#setting-variables-on-connection
+# [ ] Using variables via the IDENTIFIER function: INSERT INTO IDENTIFIER($my_table_name) (i) VALUES (42);
+# [ ] Session variable functions: https://docs.snowflake.com/en/sql-reference/session-variables#session-variable-functions
+def test_variables(conn: snowflake.connector.SnowflakeConnection):
+    with conn.cursor() as cur:
+        cur.execute("SET var1 = 1;")
+        cur.execute("SET var2 = 'hello';")
+        cur.execute('SET var3 = parse_json(\'{"k1": "v1"}\');')
+        cur.execute("SET var4 = array_construct(1, 2, 3);")
+        cur.execute("SET day = CURRENT_DATE();")
+
+        cur.execute("select $var1, $var2, $var3, $var4, $day;")
+        assert cur.fetchall() == [(1, "hello", '{"k1":"v1"}', [1, 2, 3], datetime.date.today())]
+
+        cur.execute("CREATE TABLE example (id int, name varchar);")
+        cur.execute("INSERT INTO example VALUES (10, 'hello'), (20, 'world');")
+        cur.execute("select id, name from example where name = $var2;")
+        assert cur.fetchall() == [(10, "hello")]
+
+        cur.execute("UNSET var3;")
+        try:
+            cur.execute("select $var3;")
+            raise AssertionError("Expected exception")
+        except duckdb.duckdb.InvalidInputException:
+            pass
```

Comment:
To test for an exception I'd use [pytest.raises](https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions), eg:
```suggestion
        with pytest.raises(ProgrammingError, match='Session variable '$VAR3' does not exist'):
        cur.execute("select $var3;")
```

Note also fakesnow should raise the snowflake `ProgrammingError` rather than a duckdb error, to avoid the duckdb implementation leaking out into clients.

And optionally have a similar error message to Snowflake eg: 
> Session variable '$VAR3' does not exist